jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -746,10 +746,8 @@ public static boolean interrupted() {
  * @see			Thread#start
  */
 public final boolean isAlive() {
-	synchronized (lock) {
```

Comment:
Do we have the source as to why these locks were acquired here in the first place? I suspect it has something to do with guarding the `volatile` variables.

Reply:
They seem to exist since the beginning as means to protect access to shared variables. I didn't find a special reason for their use here. Based on how on how these methods will be used, they should function fine without synchronization.

Reply:
I also checked in the extension repo code for j.l.Thread that we use for the more recent JDK versions, and `isAlive` already doesn't have the `synchronized` block.`isDead` still does though. Are you opening a patch for the extension repos as well to remove the `isDead` lock?

Reply:
> Are you opening a patch for the extension repos as well to remove the isDead lock?

Yes.